frontend/src/ResourceSearchBar.svelte

```diff
@@ -0,0 +1,130 @@
+<style>
+  select {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  input {
+    background: inherit;
+    color: inherit;
+    border: 1px solid;
+    border-bottom: 1px solid var(--main-fg-color);
+    flex: 1;
+  }
+
+  form {
+    display: inherit;
+    flex: 1;
+  }
+
+  label {
+    display: inherit;
+    flex: 1;
+  }
+
+  .searchbar {
+    display: flex;
+    align-items: left;
+    flex-grow: 1;
+    height: 2em;
+    position: sticky;
+    padding-bottom: 1em;
+  }
+
+  .resultwidgets {
+    display: flex;
+  }
+
+  .resultcount {
+    margin-top: 0;
+    margin-bottom: 0;
+    border-style: solid;
+    width: 100%;
+    padding-left: 1em;
+    padding-right: 1em;
+  }
+
+  button {
+    height: 100%;
+  }
+</style>
+
+<script>
+  import { selectedResource } from "./stores";
+
+  let searchType, searchQuery;
+  let searchTypes = ["String", "Bytes"];
+  let searchFilter = null;
+
+  export let search, searchResults;
+
+  searchResults.matches = undefined;
+  searchResults.index = 0;
+
+  let prevQuery = "";
+
+  function nextMatch() {
+    let nextIndex = searchResults.index + 1;
+    if (nextIndex >= searchResults.matches.length) {
+      nextIndex = 0;
+    }
+    searchResults = { matches: searchResults.matches, index: nextIndex };
+  }
+
+  function prevMatch() {
+    let nextIndex = searchResults.index - 1;
+    if (nextIndex < 0) {
+      nextIndex = Math.max(searchResults.matches.length - 1, 0);
+    }
+    searchResults = { matches: searchResults.matches, index: nextIndex };
+  }
```

Comment:
Another strange characteristic is the way that multiple contiguous regions can be matched multiple times. Personally, I think this behavior should be considered a bug. Even if it's intended, it's unintuitive for the user what's going on. 

Maybe the particular instance of the found string that corresponds to the latest find should be highlighted in a different color? Kind of how <kbd>Ctrl</kbd>+<kbd>f</kbd> in the browser will highlight all the instances it's found, but highlight the current one in a different color. 

https://github.com/redballoonsecurity/ofrak/assets/99368685/52f9fe00-4218-4d5a-ad00-bec16db02d93

Reply:
I did a simple implementation to try to address this, just underlining the bytes if the range they match (from the binary search) is the same index as the selected range. It sort of works, but doesn't work well for exactly the case being raised here: Multiple matching ranges overlapping, since not all of the bytes in the currently selected range are highlighted.  I would guess that the binary search finds one of the overlapping ranges instead, not prioritizing the selected one.

So I think it would actually considerably complicate the logic, unless you see another simple way to try it. I think we should leave this as future work.